relay.go

```diff
@@ -170,10 +170,11 @@ type Relayer struct {
 	// It stores remappings for all response frames read on this connection.
 	inbound *relayItems
 
-	peers   *PeerList
-	conn    *Connection
-	logger  Logger
-	pending atomic.Uint32
+	peers     *PeerList
+	conn      *Connection
+	relayConn relay.Conn // cast and allocate once
```

Comment:
Since the `conn` is a pointer already, the conversion to an interface is pretty cheap (no allocs). I'd drop the field and pass it directly